### ISSUE TITLE
Enable assigning multiple languages to a single asset

### ISSUE THREAD

**Issue by stijnvermeeren-swisstopo** [swisstopo/swissgeol-assets-suite#29]
Currently, the database schema of the Assets application only allows linking each asset with a single language.  This language is stored in the column `language_item_code` in the `asset` table, which has a foreign key to the `language_item` table. There, the following possible values are defined:

```
"language_item_code","geol_code","name","name_de","name_fr","name_rm","name_it","name_en","description","description_de","description_fr","description_rm","description_it","description_en"
"DE","No-GeolCode-specified","german","deutsch","allemand","tudestg","tedesco","german","Asset in German","Asset in Deutsch","Asset en allemand","","Elemento in tedesco","Asset in German"
"EN","No-GeolCode-specified","english","englisch","anglais","englais","inglese","english","Asset in English","Asset in Englisch","Asset en anglais","","Asset in inglese","Asset in English"
"FR","No-GeolCode-specified","french","französisch","français","franzos","francese","french","Asset in French","Asset in Französisch","Asset en français","","Elemento in francese","Asset in French"
"IT","No-GeolCode-specified","italian","italienisch","italien","talian","italiano","italian","Asset in Italian","Asset in Italienisch","Asset en italien","","Elemento in italiano","Asset in Italian"
"NUM","No-GeolCode-specified","numeric","numerisch","numérique","","numerico","numeric","Asset with numerical structure, e.g. programme code, configurations","Asset mit numerischem Aufbau, wie z.B. Programmcode, Konfigurationen","Asset à structure numérique, p. ex. code de programme, configurations","","Elemento con struttura numerica, ad es. codice programma, configurazioni","Asset with numerical structure, e.g. programme code, configurations"
"other","No-GeolCode-specified","other languages","andere Sprachen","autres langues","autras linguas","altre lingue","other languages","Asset in other languages","Asset in anderer Sprachen","Assets dans d'autres langues","","Elemento in altre lingue","Asset in other languages"
```

As we have a significant number of assets that contain a significant amount of text in several different languages, it would make more sense to enable linking multiple languages for each asset. The interpretation would be as follows:

* No languages linked to an asset
  * asset language is "unknown" / not yet defined.
* Asset language is any combination of "DE", "EN", "FR", "IT", "other"
  * Asset contains a significant amount of text in each of the selected languages.
  * "other" can stand for any natural language that is not German, English, French of Italian. For example Russian, Dutch, etc.
* Asset language "NUM" (numeric)
  * Assets that only contain numerical tables, graphs, source code, etc. (probably only for non-PDF file types)
  * Currently, we don't have any examples of such assets, but we keep this value for potential future usage
  * We currently don't see any use case for assigning both "NUM" as well as other languages to a single asset, but I still would leave this as a possibility (i.e. don't add any validation rules / database constraints that prevent this)

**ACs**

* Database schema migration
  * Add a new table, named something like `asset_language`, with columns `asset_id` (foreign key to the `asset` table) and `language_code` (foreign key to the `language_item`) table, where we can store asset-language relationships as a many-to-many relationship type.
* Database data migration
  * Migrate all the `language_item_code` values (except for "other") in the `asset` table to the new table.
    * For assets that currently have `language_item_code = "other"` in the `assets` table, no entry should be created in the new table.
* Database schema migration
  * Remove the column `language_item` from the `asset` table.
* UI changes
  * Change the create/edit asset forms, so that for each asset, any number of languages can be selected (maybe use checkboxes instead of dropdown?)
    * Remove any validation rules (the asset should also be valid, even when no languages are selected)
  * Change the "detailed information" view for a specific asset to list all linked languages as a comma-separated list
    * When no language is linked to the asset, then display "Language: unknown" in the "detailed information" for that asset
* Search functionality
  * In addition to the currently available filter buttons ("andere Sprachen", "deutsch", "englisch", "französisch", "italienisch", "numerisch"), add an additional button "unbekannt".
    * Assets that are not linked to any language, should be shown in the results, if and only if this button is selected.
  * When the user has selected one or more languages in the search filter, then in the search results, only assets are shown that are linked with at least one of these languages.
    * The shown assets may also be linked with additional languages which the user did not select.
  * Otherwise, the search behavior should remain unchanged. In particular:
    * A language filter button is enabled for a particular search query, if and only if there is at least one result that matches this language filter
    * When multiple buttons are selected, then the search results shown are the union of the results that match each individual filter.
    * By default, all language filter buttons are selected (all search results are shown, regardless of language)
* Elasticsearch
  * **To Be Defined**: is it necessary to change the structure of the Elasticsearch index?
  * We will probably also need to update all existing assets in the Elasticsearch index that used to have `language = "other"`, as they will now have `language = []` (empty list)

- [x] #78
- [x] #79
- [x] #80

**Comment by vej-ananas** [comment]
- Datenbank 1
- Backend + Unit Tests 2
- UI 3
- Search functionality 2
- Elasticsearch + Search Backend

**Comment by daniel-va** [comment]
Resolved via #78 and #79.
